bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go

```diff
@@ -112,8 +135,63 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	return allErrs
+}
+
+func (c *KubeadmConfigSpec) validateIgnition() field.ErrorList {
```

Comment:
I'm not sure a denylist approach is the right one for the webhook. The problem with the current approach is how we can keep it up-to-date / in sync with the cloud-init implementation. 

But the alternative is probably too much effort (e.g. implement an allow list of ignition-supported fields, like we currently to for mutable fields in KCP)

@fabriziopandini WDYT?